pyvista/core/dataset.py

```diff
@@ -1407,19 +1411,28 @@ def flip_normal(
             t, transform_all_input_vectors=transform_all_input_vectors, inplace=inplace
         )
 
-    def copy_meta_from(self, ido: 'DataSet'):
+    def copy_meta_from(self, ido: 'DataSet', deep: bool = True):
         """Copy pyvista meta data onto this object from another object.
 
         Parameters
         ----------
         ido : pyvista.DataSet
             Dataset to copy the metadata from.
 
+        deep : bool, optional
+            Deep or shallow copy.
+
         """
-        self._active_scalars_info = ido.active_scalars_info
-        self._active_vectors_info = ido.active_vectors_info
         self.clear_textures()
-        self._textures = {name: tex.copy() for name, tex in ido.textures.items()}
+
+        if deep:
+            self._active_scalars_info = ido.active_scalars_info.copy()
+            self._active_vectors_info = ido.active_vectors_info.copy()
+            self._textures = {name: tex.copy() for name, tex in ido.textures.items()}
+        else:
+            self._active_scalars_info = ido.active_scalars_info
+            self._active_vectors_info = ido.active_vectors_info
+            self._textures = ido.textures
```

Comment:
Sorry for nitpicking all the time, but this textures line here is not a shallow copy, it's the same reference to the original (in other words, it's not a copy at all).

For clarity, I can imagine these three cases:
```py
self._textures = ido.textures  # shared reference

self._textures = ido.textures.copy()  # shallow copy

self._textures = {name: tex.copy() for name, tex in ido.textures.items()}  # deep copy
# or
self._textures = copy.deepcopy(ido.textures)  # also deep copy
```

The functional difference is this:
1. if an item is added to `ido.textures`, it will be reflected in the reference, but not in any of the copies
2. if an existing item in `ido.textures` is modified (mutated), it will be reflected in the reference and the shallow copy, but not the deep copy.

The current version might easily be correct, I just want to make sure that it does what you want it to do.

Reply:
> this textures line here is not a shallow copy,

You're right. It's the exact same reference.

> The current version might easily be correct (in fact I think this is on purpose), I just want to make sure that it does what you want it to do.

Current implementation is correct, and like `point_data` and `cell_data`, when an array is added to the original, a shallow copy dataset will have it reflected in their corresponding dataset attributes. Since we don't have a class to abstract out the `textures`, instead of creating a shallow copy with pointers to the underlying data, we're just referencing the original. This way, operations like `clear_textures` clears the textures on the original and copied dataset because they're sharing the same reference to the same dictionary.

> Sorry for nitpicking all the time

Always appreciated. Better to have this out in the open than write bad code. It makes writing/merging these PRs take longer, but in the long run I think we'll save time. Had we considered these issues when this `copy_metadata` was implemented, we wouldn't be having to spend time to fix it.

---

I'm considering this resolved, unresolve if we're not done.

Reply:
Great, thanks!